In [2]:
import warnings
warnings.filterwarnings("ignore", category=FutureWarning)
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline

In [7]:
my_df = pd.read_csv('report1_performance_s30_b2_s2.csv', header = [0,1], index_col = 0)
my_df

MSFT                            \
                         B&H % increase Strat risk-adj % increase   
1984-01-02 to 1994-01-01        2185.43                    675.23   
1994-01-02 to 2004-01-01        1022.01                    464.30   
2004-01-02 to 2014-01-01          89.42                    199.72   
2014-01-02 to 2024-01-01        1087.38                   1272.09   

                                                                  AAPL  \
                         No. of trades Win / Loss Ratio B&H % increase   
1984-01-02 to 1994-01-01            28             1.55         157.65   
1994-01-02 to 2004-01-01            38             2.45          18.05   
2004-01-02 to 2014-01-01            34             1.27        4945.40   
2014-01-02 to 2024-01-01            40             3.00        1022.73   

                                                                  \
                         Strat risk-adj % increase No. of trades   
1984-01-02 to 1994-01-01                   -155.54            33   
1994-01-02 to 2004-01-01                   -151.27            38   
2004-01-02 to 2014-01-01                   1242.77            38   
2014-01-02 to 2024-01-01                    478.24            37   

                                                    NVDA  \
                         Win / Loss Ratio B&H % increase   
1984-01-02 to 1994-01-01             0.94           0.00   
1994-01-02 to 2004-01-01             1.11         340.59   
2004-01-02 to 2014-01-01             3.22         112.13   
2014-01-02 to 2024-01-01             2.36       11994.73   

                                                    ...            GE  \
                         Strat risk-adj % increase  ... No. of trades   
1984-01-02 to 1994-01-01                      0.00  ...            39   
1994-01-02 to 2004-01-01                   1055.19  ...            39   
2004-01-02 to 2014-01-01                     74.94  ...            38   
2014-01-02 to 2024-01-01                   1203.39  ...            33   

                                                   CMCSA  \
                         Win / Loss Ratio B&H % increase   
1984-01-02 to 1994-01-01             5.50        1425.44   
1994-01-02 to 2004-01-01             2.00         230.73   
2004-01-02 to 2014-01-01             2.17         190.99   
2014-01-02 to 2024-01-01             1.75         102.14   

                                                                  \
                         Strat risk-adj % increase No. of trades   
1984-01-02 to 1994-01-01                   1715.79            35   
1994-01-02 to 2004-01-01                    229.71            36   
2004-01-02 to 2014-01-01                    155.56            39   
2014-01-02 to 2024-01-01                    -63.93            34   

                                                      VZ  \
                         Win / Loss Ratio B&H % increase   
1984-01-02 to 1994-01-01             3.38         477.73   
1994-01-02 to 2004-01-01             1.25          81.72   
2004-01-02 to 2014-01-01             1.79         134.71   
2014-01-02 to 2024-01-01             1.00          27.65   

                                                                  \
                         Strat risk-adj % increase No. of trades   
1984-01-02 to 1994-01-01                   1270.96            37   
1994-01-02 to 2004-01-01                    -32.33            34   
2004-01-02 to 2014-01-01                    115.89            40   
2014-01-02 to 2024-01-01                      4.28            37   

                                           
                         Win / Loss Ratio  
1984-01-02 to 1994-01-01             8.25  
1994-01-02 to 2004-01-01             1.00  
2004-01-02 to 2014-01-01             1.67  
2014-01-02 to 2024-01-01             1.18  

[4 rows x 196 columns]

In [9]:
my_df['MSFT']

B&H % increase  Strat risk-adj % increase  \
1984-01-02 to 1994-01-01         2185.43                     675.23   
1994-01-02 to 2004-01-01         1022.01                     464.30   
2004-01-02 to 2014-01-01           89.42                     199.72   
2014-01-02 to 2024-01-01         1087.38                    1272.09   

                          No. of trades  Win / Loss Ratio  
1984-01-02 to 1994-01-01             28              1.55  
1994-01-02 to 2004-01-01             38              2.45  
2004-01-02 to 2014-01-01             34              1.27  
2014-01-02 to 2024-01-01             40              3.00

In [20]:
risk_adj_gains = my_df.xs(key = 'Strat risk-adj % increase', level = 1, axis = 1)
mean_risk_adj_gains = risk_adj_gains.mean().to_frame()
mean_risk_adj_gains

0
MSFT    652.8350
AAPL    353.5500
NVDA    583.3800
AMZN    310.7375
GOOG    203.7950
GOOGL   198.1925
META    107.8275
LLY     326.2100
TSLA    239.4275
AVGO    113.2900
V       250.5050
JPM     316.8925
UNH    2133.0525
WMT     566.8250
MA      486.8050
XOM     245.7825
JNJ     426.3350
PG      293.6775
HD      809.4600
COST    262.5125
MRK     428.1575
ABBV     77.0075
ORCL    615.3950
CRM     420.8600
CVX     204.8350
AMD     100.5875
BAC     267.4375
KO      259.8225
NFLX    459.3325
ADBE    279.5475
ACN     211.3425
PEP     226.1700
LIN     169.2900
TMO     446.5350
MCD     261.9925
ABT     326.5875
DIS     172.2850
CSCO    539.6500
TMUS    156.8750
WFC     380.3875
DHR     579.8925
INTU    422.9025
INTC    158.7925
QCOM    623.9150
AMAT   1619.3225
IBM      23.2475
GE      257.7225
CMCSA   509.2825
VZ      339.7000

In [21]:
bah_gains = my_df.xs(key = 'B&H % increase', level = 1, axis = 1)
mean_bah_gains = bah_gains.mean().to_frame()
mean_bah_gains

0
MSFT   1096.0600
AAPL   1535.9575
NVDA   3111.8625
AMZN   1223.5900
GOOG    284.0800
GOOGL   282.4900
META    125.3575
LLY     541.7550
TSLA    614.3850
AVGO    654.9900
V       143.8400
JPM     207.3075
UNH    1240.5725
WMT     529.7175
MA      561.2300
XOM     243.4000
JNJ     307.8800
PG      279.6125
HD      752.0675
COST    301.3800
MRK     319.7575
ABBV    103.4650
ORCL    701.3800
CRM     514.8900
CVX     222.0800
AMD     954.1375
BAC     213.1250
KO      422.4475
NFLX    455.6675
ADBE    601.2725
ACN     198.7825
PEP     424.1875
LIN     255.8875
TMO     362.6875
MCD     334.4475
ABT     311.4050
DIS     410.2700
CSCO    844.8750
TMUS     94.5575
WFC     316.5625
DHR     676.5925
INTU    433.2575
INTC    341.2150
QCOM    574.1850
AMAT    639.8025
IBM     197.6575
GE      198.0400
CMCSA   487.3250
VZ      180.4525

In [22]:
comparison = mean_risk_adj_gains - mean_bah_gains
comparison

0
MSFT   -443.2250
AAPL  -1182.4075
NVDA  -2528.4825
AMZN   -912.8525
GOOG    -80.2850
GOOGL   -84.2975
META    -17.5300
LLY    -215.5450
TSLA   -374.9575
AVGO   -541.7000
V       106.6650
JPM     109.5850
UNH     892.4800
WMT      37.1075
MA      -74.4250
XOM       2.3825
JNJ     118.4550
PG       14.0650
HD       57.3925
COST    -38.8675
MRK     108.4000
ABBV    -26.4575
ORCL    -85.9850
CRM     -94.0300
CVX     -17.2450
AMD    -853.5500
BAC      54.3125
KO     -162.6250
NFLX      3.6650
ADBE   -321.7250
ACN      12.5600
PEP    -198.0175
LIN     -86.5975
TMO      83.8475
MCD     -72.4550
ABT      15.1825
DIS    -237.9850
CSCO   -305.2250
TMUS     62.3175
WFC      63.8250
DHR     -96.7000
INTU    -10.3550
INTC   -182.4225
QCOM     49.7300
AMAT    979.5200
IBM    -174.4100
GE       59.6825
CMCSA    21.9575
VZ      159.2475

In [23]:
comparison.mean()

0   -130.775102
dtype: float64

### On average, the risk adjusted returns of this startegy are 130.775 percentage points lower than those of the Buy and Hold Stratgey